torch/_functorch/aot_autograd.py

```diff
@@ -1204,6 +1206,14 @@ def inner(*flat_args):
             [x for x in input_info if x.mutates_data or x.mutates_metadata]
         )
 
+        def clone(x):
+            if isinstance(x, NestedTensor):
+                should_prop = any(any(is_symbolic(s) for s in _inner.shape) for _inner in (x._values, x._offsets))
+                ragged_size = x._size[x._ragged_idx] if should_prop else None
+                return NestedTensor(x._values, x._offsets.clone(), ragged_size=ragged_size)
+            else:
+                return x
```

Comment:
This feels a little weird, but here's how I'm trying to rationalize it.

This feels weird because normally if I clone offsets in user code, I would run into the problem where if I query for the "singleton int" of that offset it would be different. 

In this case, I have some s0 corresponding to a real value (hint) and its sympy counterpart SingletonInt (a subclass of AtomicExpr). That hint corresponds to a particular non-fake offsets, which makes it feel weird, but maybe it doesn't matter, and here's why:

Since everything is symbolic, all I have to work with are fake tensors, and it is not verifiable whether that hint corresponds to any fake tensor. This is the same reason why we propagate ragged_size everywhere anyway. Therefore in the symbolic world, no matter what fake offsets you are holding you are you free to pretend to actually have the "id" of any other offsets.

Reply:
(this function probably belongs in nested_tensor.py, if we do indeed want something like it)

Reply:
Hmm one question - Is it the case that this entire problem came around because of the fact that we're tracing a joint graph where some of the primals and tangents are the **same** tensor (offsets shows up as both `primals_4` and `tangents_2`)? If so, what do you think of the following:

```
def view_avoid_dupes_with_primals(t):
    if isinstance(t, Tensor) and is_traceable_wrapper_subclass(t):
        return transform_subclass(t, lambda _, inner_t: view_avoid_dupes_with_primals(inner_t))
    if isinstance(t, Tensor):
        return t.view(t.shape)
    return t

traced_tangents = pytree.tree_map(view_avoid_dupes_with_primals, traced_tangents)
```

This will just do a (sort of no-op) view on every inner-tensor-tangent to the joint graph, that way we're guaranteed we have no duplicates across the primals and tangents.
(1) no need to special-case NestedTensor (or how that the subclass has a reasonable implementation of clone), since it should always be safe to view the inner tensors of a subclass
(2) I did view here vs. clone - I guess it doesn't matter too much, although clone maybe feels like overkill since all we really care about is tensor identity

Reply:
Yes that is correct, and adding the more general support sounds good to me! It is also cool that this would support nesting subclasses.